level_tests.py

```diff
@@ -21,5 +21,6 @@ def test__get_next_event__moves_out_of_order__throws_exception(self, mock_level_
         mock_level_file_open.return_value.__next__ = lambda self: next(iter(self.readline, ''))
 
         filename = "filename.txt"
-        # mock_level_file_open.assert_called_once_with(filename, 'r') # I don't understand this but it's failing when I'm faily sure it shouldn't be
+        # mock_level_file_open.assert_called_once_with(filename, 'r')
```

Comment:
Yo, uncomment this and put it after your assertRaises. You are asserting that the mock has been called before you've constructed the Level object (i.e. the file open hasn't been called yet)